yarn-project/cli/src/cmds/validator_keys/valkeys.test.ts

```diff
@@ -0,0 +1,368 @@
+      const { validators, summaries } = buildValidatorEntries({
+        validatorCount: 1,
+        publisherCount: 3,
+        baseAccountIndex: 0,
+        baseAddressIndex: 0,
+        mnemonic: TEST_MNEMONIC,
+        feeRecipient: ('0x' + '33'.repeat(32)) as unknown as AztecAddress,
+      });
+      const v = validators[0] as any;
+      expect(Array.isArray(v.publisher)).toBe(true);
+      expect(v.publisher.length).toBe(3);
+      expect(summaries[0].publisherEth!.length).toBe(3);
+    });
+  });
+
+  describe('keystore output path resolution and writing', () => {
+    it('resolves explicit file path relative to provided dataDir', async () => {
+      const { outputPath, resolvedDir } = await resolveKeystoreOutputPath(tmp, 'keys.json');
+      expect(outputPath).toBe(join(tmp, 'keys.json'));
+      expect(resolvedDir).toBe(tmp);
+    });
+
+    it('finds the next available keyN.json when file is omitted', async () => {
+      // pre-create key1.json to force selection of key2.json
+      const first = join(tmp, 'key1.json');
+      writeFileSync(first, '{}', { encoding: 'utf-8' });
+      const { outputPath } = await resolveKeystoreOutputPath(tmp);
+      expect(outputPath.endsWith('key2.json')).toBe(true);
+    });
+
+    it('writes a keystore file with pretty JSON', async () => {
+      const path = join(tmp, 'out.json');
+      const obj = { schemaVersion: 1, validators: [] };
+      await writeKeystoreFile(path, obj);
+      const raw = readFileSync(path, 'utf-8');
+      expect(JSON.parse(raw)).toEqual(obj);
+    });
+  });
+
+  describe('summary and json logging helpers', () => {
+    it('prints summaries with expected structure', () => {
+      const logs: string[] = [];
+      const log = (s: string) => logs.push(s);
+      const summaries = [
+        {
+          attesterEth: '0xaaaa',
+          attesterBls: '0x1111',
+          publisherEth: ['0xpub1', '0xpub2'],
+        },
+      ];
+      logValidatorSummaries(log, summaries);
+      expect(logs.length).toBe(1);
+      const out = logs[0];
+      expect(out).toContain('acc1:');
+      expect(out).toContain('attester:');
+      expect(out).toContain('eth: 0xaaaa');
+      expect(out).toContain('bls: 0x1111');
+      expect(out).toContain('- 0xpub1');
+      expect(out).toContain('- 0xpub2');
+    });
+  });
+
+  describe('newValidatorKeystore', () => {
+    it('creates a keystore file and logs a summary', async () => {
+      const path = join(tmp, 'created.json');
+      const logs: string[] = [];
+      const log = (s: string) => logs.push(s);
+      await newValidatorKeystore(
+        {
+          dataDir: tmp,
+          file: 'created.json',
+          count: 2,
+          publisherCount: 1,
+          mnemonic: TEST_MNEMONIC,
+          feeRecipient: ('0x' + '44'.repeat(32)) as unknown as AztecAddress,
+        },
+        log,
+      );
+      expect(existsSync(path)).toBe(true);
+      const json = JSON.parse(readFileSync(path, 'utf-8')) as any;
```

Comment:
With all of these tests that write files and then read them back, can we use the `node-leystores` methods for reading and validating the schema?